awx/main/tasks/jobs.py

```diff
@@ -784,12 +784,16 @@ def sync_and_copy(self, project, private_data_dir, scm_branch=None):
             project_path = project.get_project_path(check_if_exists=False)
             if project.scm_type == 'git' and (scm_branch and scm_branch != project.scm_branch):
                 if os.path.exists(project_path):
-                    git_repo = git.Repo(project_path)
-                    if git_repo.head.is_detached:
-                        is_commit = True
-                        original_branch = git_repo.head.commit
-                    else:
-                        original_branch = git_repo.active_branch
+                    try:
+                        git_repo = git.Repo(project_path)
+                        if git_repo.head.is_detached:
+                            is_commit = True
+                            original_branch = git_repo.head.commit
+                        else:
+                            original_branch = git_repo.active_branch
+                    except git.exc.InvalidGitRepositoryError:
+                        logger.error(f"Invalid git repository at {project_path}. Removing directory and continuing.")
```

Comment:
This log would lose too much information. I'd prefer convert `logger.error` to `logger.exception`. Because, diving into questions about what this fixes, the answers were too broad. It could be because the entire directory contents were deleted, but it could be due to a ratelimiting error leaving corrupted data. Not having the specific error pinned down in the planning of this, we definitely need to at absolute minimum have that information in the logs when this happens.